### DISCUSSION TITLE
Talkgroup Organization

### DISCUSSION THREAD

**Discussion by RadiosPRN** [chuot/rdio-scanner#170] [Q&A] [answered]
Is there an easy way to organize talkgroups on the configuration page? Currently I have it set to save new talkgroups and then occasionally go back through and name them and edit the details. I have it organized with counties/agencies grouped together in order and the problem is when a new talkgroup needs to go near the bottom of the list it takes a while to drag it down there. Thought maybe there is an easier way to do that.

**Comment by chuot** [comment]
How many talkgroups you have in that system?

There's also the sort-talkgroup option you could use. It sorts talkgroups per talkgroupId within the same system and it could do the trick for you if the talkgroup mapping on that system matches counties/agencies. I know it's the case for some, but not always.

**Reply by RadiosPRN** [reply]
I would say the system has several thousand talkgroups, my instance only has between 1 and 2 hundred as of now but it keeps growing.

The sort option would be better than nothing because some of the ID's are in order but not nearly all of them. Is there an option to increase the scroll speed when dragging talkgroups down the list? That would help speed it up significantly. Or maybe have multiple columns or the ability to group the ID's? 

Another option I thought would be helpful is if there is a way to export the data to Excel where I could manually add talkgroups and edit details whenever there are changes to be made and just import the updated spreadsheet each time.

**Reply by chuot** [reply]
There's the config export you could use, but that's in json format.

Or, maybe you could keep a csv file up to date with the correct order which you can then reimport to the system with the tools from the admin dashboard. Make sure not to change the TGID.

**Reply by chuot** [reply]
I'm still gonna look for the scroll speed when draging. Maybe the angular-material has something configurable for this.

**Reply by RadiosPRN** [reply]
Is there a way to import all fields from the CSV file? When I try to do that it will import the ID, Label and Description. Is there a way to also import Group, Tag, and LED Color? If I am able to import all the fields from CSV that would solve my problem.

**Reply by chuot** [reply]
Yes, use the trunk-recorder format when you import the CSV file from the admin dashboard. The CSV format is described [here](https://github.com/robotastic/trunk-recorder/blob/dd3e6bfe1c74a7846d6e50c6e98024044d7b64f3/trunk-recorder/talkgroups.cc#L85-L94).

**Reply by RadiosPRN** [reply]
Perfect! That's what I was needing. The only thing not on there that I use is the light color, I am assuming that is not an importable option? It's not a big deal, the most important things are there.

**Comment by chuot** [comment, marked as answer]
So it looks like newer versions of Angular Material have changed the default drag and drop scroll value to 2 pixels, which is a nightmare. I didn't realize it until I tried. It wasn't that slow with previous versions of Rdio Scanner, but I'm not sure exactly when this problem appeared. Luckily there is now a configurable option that I use to change the scroll speed to 64 pixels. Therefore, your issue will be resolved in the upcoming v6.3.

**Reply by RadiosPRN** [reply]
Thanks so much!

**Reply by wmg819** [reply]
I have found that when clicking on a talk group to drag it to a new position using the page down key on the keyboard while holding the talk group will allow it to move down the list faster.
